holidays/calendars.py

```diff
@@ -942,3 +945,1420 @@ def ok_phansa_date(self, year: int) -> Optional[date]:
+            1991: ((AUG, 16),),
+            1992: ((AUG, 4),),
+            1993: ((JUL, 25),),
+            1994: ((JUL, 14),),
+            1995: ((JUL, 4),),
+            1996: ((JUN, 22),),
+            1997: ((JUN, 11),),
+            1998: ((MAY, 31),),
+            1999: ((MAY, 21),),
+            2000: ((MAY, 10),),
+            2001: ((APR, 30),),
+            2002: ((APR, 19),),
+            2003: ((APR, 8),),
+            2004: ((MAR, 27),),
+            2005: ((MAR, 16),),
+            2006: ((MAR, 6),),
+            2007: ((FEB, 24),),
+            2008: ((FEB, 13),),
+            2009: ((FEB, 1),),
+            2010: ((JAN, 21),),
+            2011: ((JAN, 10), (DEC, 31)),
+            2012: ((DEC, 19),),
+            2013: ((DEC, 9),),
+            2014: ((NOV, 28),),
+            2015: ((NOV, 18),),
+            2016: ((NOV, 6),),
+            2017: ((OCT, 26),),
+            2018: ((OCT, 15),),
+            2019: ((OCT, 5),),
+            2020: ((SEP, 23),),
+            2021: ((SEP, 13),),
+            2022: ((SEP, 2),),
+            2023: ((AUG, 22),),
+            2024: ((AUG, 10),),
+            2025: ((JUL, 31),),
+            2026: ((JUL, 20),),
+            2027: ((JUL, 10),),
+            2028: ((JUN, 29),),
+            2029: ((JUN, 18),),
+            2030: ((JUN, 7),),
+            2031: ((MAY, 27),),
+            2032: ((MAY, 15),),
+            2033: ((MAY, 5),),
+            2034: ((APR, 25),),
+            2035: ((APR, 14),),
+            2036: ((APR, 3),),
+            2037: ((MAR, 23),),
+            2038: ((MAR, 12),),
+            2039: ((MAR, 1),),
+            2040: ((FEB, 19),),
+            2041: ((FEB, 7),),
+            2042: ((JAN, 28),),
+            2043: ((JAN, 17),),
+            2044: ((JAN, 7), (DEC, 26)),
+            2045: ((DEC, 15),),
+            2046: ((DEC, 4),),
+            2047: ((NOV, 24),),
+            2048: ((NOV, 12),),
+            2049: ((NOV, 2),),
+            2050: ((OCT, 22),),
+            2051: ((OCT, 11),),
+            2052: ((SEP, 29),),
+            2053: ((SEP, 18),),
+            2054: ((SEP, 8),),
+            2055: ((AUG, 29),),
+            2056: ((AUG, 17),),
+            2057: ((AUG, 6),),
+            2058: ((JUL, 26),),
+            2059: ((JUL, 16),),
+            2060: ((JUL, 4),),
+            2061: ((JUN, 24),),
+            2062: ((JUN, 13),),
+            2063: ((JUN, 3),),
+            2064: ((MAY, 22),),
+            2065: ((MAY, 11),),
+            2066: ((APR, 30),),
+            2067: ((APR, 20),),
+            2068: ((APR, 8),),
+            2069: ((MAR, 29),),
+            2070: ((MAR, 19),),
+            2071: ((MAR, 8),),
+            2072: ((FEB, 25),),
+            2073: ((FEB, 13),),
+            2074: ((FEB, 2),),
+            2075: ((JAN, 23),),
+            2076: ((JAN, 12),),
+            2077: ((JAN, 1),),
+        },
+    }
+
+    @staticmethod
+    def islamic_holiday_date(year: int, hol_type: str) -> Set[date]:
```

Comment:
The other calendars use a method per holiday approach. I believe this class should copy that behaviour too.
It's totally fine to submit it as is for now.

Reply:
I don't know how to implement `_add_islamic_calendar_holiday()` functionality in this case (for multi-day holidays). 🤷‍♂️

Reply:
The holidays dates would be used in the calendar methods directly (not via `_add_islamic_calendar_holiday`)